File: python/merge/Merger_epub_pdf.py
from aspose.pdf import (
    SaveFormat,
    EpubLoadOptions,
    Document
)

import aspose.pdf.LoadOptions as LoadOptions


def epub_to_pdf():
    path_source1 = "../../TestData/test.epub"
    path_source2 = "../../TestData/Second/test.epub"

    opt1 = EpubLoadOptions()
    # use algorithm to prevent content to be truncated
    opt1.page_size_adjustment_mode = \
        LoadOptions.pagesizeAdjustmentModes.ENLARGE_REQUIRED_VIEWPORT_WIDTH_AND_DO_CONVERSION_AGAIN
    # usage of margins area during conversion
    opt1.margins_area_usage_mode = \
        LoadOptions.MarginsAreaUsageModes.PUT_CONTENT_ON_MARGIN_AREA_IF_NECESSARY

    # epub files can be parsed and loaded as Aspose Document
    first_doc = Document(path_source1, opt1)

    opt2 = EpubLoadOptions
    # use algorithm to prevent content to be truncated
    opt1.page_size_adjustment_mode = \
        LoadOptions.pagesizeAdjustmentModes.ENLARGE_REQUIRED_VIEWPORT_WIDTH_AND_DO_CONVERSION_AGAIN
    # usage of margins area during conversion
    opt1.margins_area_usage_mode = \
        LoadOptions.MarginsAreaUsageModes.PUT_CONTENT_ON_MARGIN_AREA_IF_NECESSARY
    second_doc = Document(path_source2, opt2)

    # create empty pdf document
    output_doc = Document()

    # set less memory usage with unload instead of fast performance
    output_doc.enable_object_unload = True

    for page in first_doc.pages:
        # add page from one document to another directly
        output_doc.pages.add(page)

    for page in second_doc.pages:
        # add page from one document to another directly
        output_doc.pages.add(page)

    # save result pdf to file
    output_doc.save("test.pdf", SaveFormat.Pdf)

File: cpp/merge/Merger_epub_pdf.cpp
#include "Aspose.PDF.Cpp/LoadOptions.h"
#include "Aspose.PDF.Cpp/EpubLoadOptions.h"
#include "Aspose.PDF.Cpp/Facades/Algorithm.h"
#include "Aspose.PDF.Cpp/Devices/Margins.h"
#include "Aspose.PDF.Cpp/Document.h"
#include "Aspose.PDF.Cpp/Page.h"
#include "Aspose.PDF.Cpp/PageCollection.h"
#include "Aspose.PDF.Cpp/SaveFormat.h"

using namespace System;
using namespace Aspose::Pdf;

void epub_to_pdf()
{
    String pathSource1 = u"../../TestData/test.epub";
    String pathSource2 = u"../../TestData/Second/test.epub";

    auto opt1 = MakeObject<EpubLoadOptions>();
    // use algorithm to prevent content to be truncated
    opt1->PageSizeAdjustmentMode = Aspose::Pdf::LoadOptions::PageSizeAdjustmentModes::EnlargeRequiredViewportWidthAndDoConversionAgain;
    // usage of margins area during conversion
    opt1->MarginsAreaUsageMode = Aspose::Pdf::LoadOptions::MarginsAreaUsageModes::PutContentOnMarginAreaIfNecessary;

    // epub files can be parsed and loaded as Aspose Document
    System::SharedPtr<Document> firstDoc = MakeObject<Document>(pathSource1, opt1);

    auto opt2 = MakeObject<EpubLoadOptions>();
    // use algorithm to prevent content to be truncated
    opt2->PageSizeAdjustmentMode = Aspose::Pdf::LoadOptions::PageSizeAdjustmentModes::EnlargeRequiredViewportWidthAndDoConversionAgain;
    // usage of margins area during conversion
    opt2->MarginsAreaUsageMode = Aspose::Pdf::LoadOptions::MarginsAreaUsageModes::PutContentOnMarginAreaIfNecessary;

    System::SharedPtr<Document> secondDoc =MakeObject<Document>(pathSource2, opt2);

    // create empty pdf document
    System::SharedPtr<Document> outputDoc = MakeObject<Document>();

    // set less memory usage with unload instead of fast performance
    outputDoc->set_EnableObjectUnload(true);

    for (int i = 1; i < firstDoc->get_Pages()->get_Count(); i++)
    {
        auto page = firstDoc->get_Pages()->idx_get(i);
        // add page from one document to another directly
        outputDoc->get_Pages()->CopyPage(page);
    }

    for (int i = 1; i < secondDoc->get_Pages()->get_Count(); i++)
    {
        auto page = secondDoc->get_Pages()->idx_get(i);
        // add page from one document to another directly
        outputDoc->get_Pages()->CopyPage(page);
    }

    // save result pdf to file
    outputDoc->Save(u"test.pdf", SaveFormat::Pdf);
}
